src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -136,19 +158,125 @@ SetInteriorVehicleDataRequest::SetInteriorVehicleDataRequest(
 
 SetInteriorVehicleDataRequest::~SetInteriorVehicleDataRequest() {}
 
+const std::string LightName(const smart_objects::SmartObject& light_name) {
+  const char* name;
+  const bool ok = NsSmartDeviceLink::NsSmartObjects::
+      EnumConversionHelper<mobile_apis::LightName::eType>::EnumToCString(
+          static_cast<mobile_apis::LightName::eType>(light_name.asUInt()),
+          &name);
+  return ok ? name : "unknown";
+}
+
+bool CheckLightDataByCapabilities(
+    const smart_objects::SmartObject& capabilities,
+    const smart_objects::SmartObject& light_data) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::map<std::string, std::string> lightCapsMapping = {
+      {message_params::kId, strings::kName},
+      {message_params::kDensity, strings::kDensityAvailable},
+      {message_params::kSRGBColor, strings::kSRGBColorSpaceAvailable}};
+  auto it = light_data.map_begin();
+  for (; it != light_data.map_end(); ++it) {
+    if (message_params::kStatus == it->first ||
+        message_params::kId == it->first) {
+      continue;
+    }
+    const std::string& caps_key = lightCapsMapping[it->first];
+    LOG4CXX_DEBUG(logger_,
+                  "Checking request parameter "
+                      << it->first << " with capabilities. Appropriate key is "
+                      << caps_key);
+
+    if (!capabilities.keyExists(caps_key)) {
+      LOG4CXX_DEBUG(logger_,
+                    "Capability "
+                        << caps_key
+                        << " is missed in RemoteControl capabilities");
+      return false;
+    }
+    if (!capabilities[caps_key].asBool()) {
+      LOG4CXX_DEBUG(logger_,
+                    "Capability "
+                        << caps_key
+                        << " is switched off in RemoteControl capabilities");
+      return false;
+    }
+  }
+  return true;
+}
+
+bool CheckLightNameByCapabilities(
+    const smart_objects::SmartObject& capabilities_status,
+    const smart_objects::SmartObject& light_data) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  auto it = capabilities_status.asArray()->begin();
+  for (; it != capabilities_status.asArray()->end(); ++it) {
+    const smart_objects::SmartObject& so = *it;
+    const int64_t current_id = so[message_params::kName].asInt();
+    if (current_id == light_data[message_params::kId].asInt()) {
+      return CheckLightDataByCapabilities(so, light_data);
+    }
+  }
+  LOG4CXX_DEBUG(logger_, "There is no such light name in capabilities");
+  return false;
+}
+
 bool CheckControlDataByCapabilities(
     const smart_objects::SmartObject& module_caps,
     const smart_objects::SmartObject& control_data) {
+  LOG4CXX_AUTO_TRACE(logger_);
   std::map<std::string, std::string> mapping =
       GetModuleDataToCapabilitiesMapping();
   const smart_objects::SmartObject& capabilities_status = module_caps[0];
```

Comment:
Just pass in an individual module from the start:
```
const smart_objects::SmartObject& capabilities_status = module_caps[0];
...
is_module_data_valid =
            CheckControlDataByCapabilities(caps, module_data[param.first][0]);
```

Reply:
@jacobkeeler fixed 8b596323783b077cfea212fbb00bd0cd161f0481